fedimint-client/src/db.rs

```diff
@@ -480,6 +475,13 @@ pub async fn apply_migrations_client(
         .await
         .is_none();
 
+    let last_key_value = migrations.last_key_value();
+    let target_version = if let Some((last_key, _)) = last_key_value {
+        DatabaseVersion(last_key.0 + 1)
+    } else {
+        DatabaseVersion(0)
+    };
+
```

Comment:
Make a function (possibly generic of V, so can be re-used between server & client). In that function verify that migrations are contiguous.

Reply:
Will make a function.

I actually don't think we need to assert that migrations are continuous. `apply_migrations` and `apply_migrations_client` already skip over missing migrations, and since we're using the last key here (and not the size of the map), we can handle non-continuous migrations.

Reply:
@m1sterc001guy Missing versions in db migration list does not seem useful for anything. It is probably a programming mistake that if not cough automatically, might get unnoticed.

Reply:
That is true, I also don't think it hurts too much. I can add the contiguous check, just wanted to point out that I don't think its strictly necessary. I'll add a comment.